### PR TITLE
Feature/improve decoder bunq/sdk java#35

### PR DESCRIPTION
**Pull request by OGKevin**
Closes bunq/sdk_java#51
Closes bunq/sdk_java#35

### CODE REVIEW

File: src/main/java/com/bunq/sdk/json/AnchorObjectAdapter.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+public class AnchorObjectAdapter implements JsonDeserializer<AnchorObjectInterface> {
+  @Override
+  public AnchorObjectInterface deserialize(JsonElement json, Type typeOfT, JsonDeserializationContext context) throws
+      JsonParseException {
+    AnchorObjectInterface model = new Gson().fromJson(json, typeOfT);
+
+
+    if (model.isAllFieldNull()) {
+
```
Comment (OGKevin) [inline src/main/java/com/bunq/sdk/json/AnchorObjectAdapter.java]: Remove line

File: src/main/java/com/bunq/sdk/json/AnchorObjectAdapter.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      Field[] allFields = model.getClass().getDeclaredFields();
+
+      for(Field field : allFields) {
+
+        if (!BunqModel.class.isAssignableFrom(field.getType())) {
+
```
Comment (OGKevin) [inline src/main/java/com/bunq/sdk/json/AnchorObjectAdapter.java]: Bruv ?

File: src/main/java/com/bunq/sdk/json/AnchorObjectAdapter.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+      JsonParseException {
+    AnchorObjectInterface model = new Gson().fromJson(json, typeOfT);
+
+
+    if (model.isAllFieldNull()) {
+
+      Field[] allFields = model.getClass().getDeclaredFields();
+
+      for(Field field : allFields) {
+
```
Comment (OGKevin) [inline src/main/java/com/bunq/sdk/json/AnchorObjectAdapter.java]: This line can also go as well

File: src/main/java/com/bunq/sdk/model/core/BunqModel.java [hunk 4]
```diff
@@ -34,6 +35,13 @@
   protected BunqModel() {
   }
 
+  public static <T> T fromJsonReader(Class<T> tClass, JsonReader reader) {
```
Comment (patrickdw1991) [inline src/main/java/com/bunq/sdk/model/core/BunqModel.java]: No empty docblock?

**Comment by OGKevin** [comment]
@patrickdw1991 All yours please 👀

**Comment by OGKevin** [comment]
@andrederoos All yours

**Pull request merged**